interview-questions.md

```diff
@@ -8,32 +8,32 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Compare and contrast arrays and hashes in Ruby. Include at least two differences and two similarities.
 
-Your answer:
+Your answer: Hash is a collection of key and value pairs. Array is an ordered collection of objects. 
 
-Researched answer:
+Researched answer: A Ruby hash is a dictionary-like container for unique keys and their corresponding values. We can implement operations on a hash using CRUD. Both hash and arrays can be nested. 
 
 2. What does it mean to implement a TTD workflow?
 
-Your answer:
+Your answer: It is how we conduct quality assurance on our code. We only write new code when automated test has failed. 
 
-Researched answer:
+Researched answer: Test-driven development (TDD) is a software development practice that emphasizes writing tests before writing the actual code. It follows a cyclical process of writing a failing test, writing the minimum code to make the test pass, and then refactoring the code.
 
 3. Why would super be used in a Ruby class?
 
-Your answer:
+Your answer: Super is used to call on the initialize method in the superclass. 
 
-Researched answer:
+Researched answer: The super method is used in the subclass. After invoking super the subclass will have access to the instance variables within that method. Instance variables are not inherited since instance variables are local to a specific instance of a class. Via super, you are allowed to borrow instance variables from the parent.
```

Comment:
Your researched answers are pulled directly from the internet and not in your own words.  Please update this so that all the researched answers are in your own words.